docs/source/advanced_config.rst

```diff
@@ -6,17 +6,64 @@ Advanced Config
 
 Custom Steps
 ============
-coming soon
+If you need a custom build step, you can create a subclass of the :class:`~fab.steps.Step` class.
 
-Metrics
-=======
-coming soon
+Fab includes some examples of a custom step. A simple example was created for building JULES.
+The :class:`~fab.steps.root_inc_files.RootIncFiles` step copies all `.inc` files in the source tree
+into the root of the source tree, to make subsequent preprocessing flags easier to configure.
+
+That was a simple example that didn't need to interact with the :term:`Artefact Store`.
+If you need to insert a custom step between two Fab steps, you may need your custom step to create
+an :term:`Artefact Collection`, and to configure the subsequent step to use your new collection.
```

Comment:
Well the custom step _could_ overwrite the artefact collection it reads with the artefacts it creates, to avoid having to tell the subsequent step where to find them, but that seems less desirable. I think the act of inserting a step, and therefore a new artefact collection, does fundamentally mean you're changing the flow of data between steps, i.e what the next step ingests. Happy to chat further, of course.